core/src/main/java/dk/alexandra/fresco/framework/configuration/NetworkConfigurationImpl.java

```diff
@@ -41,5 +45,17 @@ public String toString() {
         + parties + "]";
   }
 
+  /**
+   * Verifies that all party addresses are unique.
+   */
+  private static void checkAddressesUnique(Map<Integer, Party> parties) {
+    Set<String> addresses = new HashSet<>();
+    for (Party party : parties.values()) {
+      addresses.add(party.getHostname() + " " + party.getPort());
```

Comment:
This will work, but I would probably use a `Set<Pair<String, Integer>>` instead. Seems more clear to me, and should be more efficient (String concatenation is slow).